### DISCUSSION TITLE
whereRelationIn

### DISCUSSION THREAD

**Discussion by irakliShengelia** [laravel/framework#46224] [Ideas]
Lately I've come to find that in our app we often query for a relationship column being in a set of values. I think it would be useful (and prettier than whereHas/[whereRelation with callback](https://github.com/laravel/framework/pull/42491)) to have a whereRelationIn method, something along the lines of the below:

```
public function whereRelationIn(string $relation, string $column, array $values)

{

	return $this->whereHas($relation, function ($query) use ($column, $values) {
	
		$query->whereIn($column, $values);
	
	});

}
```

I'm committed [alongside my colleagues] to writing the necessary code, if the idea gets traction.

What do you guys think?

-------------
I also saw this thread which ended with 'whereRelation' suggestion, so I think there are devs who could benefit with this functionality

**Comment by anilkumarthakur60** [comment]
```
 public function whereRelation($relation, $column, $operator = null, $value = null)
    {
        return $this->whereHas($relation, function ($query) use ($column, $operator, $value) {
            if ($column instanceof Closure) {
                $column($query);
            } else {
                $query->where($column, $operator, $value);
            }
        });
    }
```
you can pass the column as closure and can resolve your  issue/problem

**Comment by netorochaa** [comment]
many times I needed something like this, much cleaner than using a closure
